### DISCUSSION TITLE
Can I assume that it is safe to use the same user login concurrently from several browsers?

### DISCUSSION THREAD

**Discussion by johann-petrak** [GateNLP/gate-teamware#385] [Q&A] [answered]
Is it possible and safe to have the same user log in from several browser sessions or machines and annotate at the same time? 
Alternately is there a simple way to divide the remaining documents to be annotated by annotator A among 2 or more annotators (one of which may still be A) if it turns out after setting up everything and starting annotating that the workload would be too much for that single annotator?

**Comment by ianroberts** [comment]
Each user is assigned one document at a time, with the assignment recorded server side in the database, so if you logged in twice in two separate browsers both sessions would see the same document. And I’m not sure who would win or what would happen if both sessions tried to submit the same document, whether the first one wins, the second one overwrites the first, or something else…

But equally the documents are not pre-assigned to particular annotators - when you complete one document the system looks for any other document that you have not already done and that still needs further annotation. So if you add more annotators to the partially completed project the remaining tasks will naturally share out across them and all the existing annotators.

**Reply by ianroberts** [reply]
> I’m not sure who would win or what would happen if both sessions tried to submit the same document

Having checked the code I believe the second one would overwrite the first. Which is probably wrong - I’d have expected an error if I try to submit the same form twice.

**Reply by johann-petrak** [reply]
Ah ok, thanks, so this would then not be an option. 
Is there any other simple strategy to divide the remaining work among the original annotator and one or more others?

**Reply by ianroberts** [reply]
Add more annotators to the project.

There's no real concept of the workload "belonging" to a particular annotator.  Any annotator who is assigned to a project can keep asking for documents to annotate and the system will continue assigning them documents until one of the following conditions is satisfied:

1. This annotator has already annotated x% of the whole corpus, as set in the "max proportion" setting in the project configuration, or
2. All of the documents that this annotator has not already seen have already been (or are already being) annotated by the requisite number of other annotators.

If you add another annotator to the project then they will be treated the same as the existing annotators with respect to these rules.  Of course the new annotator will be starting from a blank slate so it'll take them longer to hit condition 1.  And (if your project asks for multiple annotations per document) there's nothing to stop the new annotator from being given documents that the existing slow annotator has already completed.  This latter case will be less likely in future versions of Teamware, given my latest PR #384 that prefers to hand out documents that have been seen by fewer people, but it remains a possibility.

If you're asking for a way to treat the new annotator B and an existing annotator A as a "team" in the sense that they will share a single x% quota and they won't both be given any of the same documents to annotate, then no, that's not something Teamware supports.

**Reply by johann-petrak** [reply]
Hmm so if I understand correctly, if the maximum number of annotations per document is changed to 1 but there are more than one annotator, they will all get different documents to annotate until all documents have one annotation?
I guess that is what I want then .. thanks!

**Reply by ianroberts** [reply, marked as answer]
Yes, if your project is not doing multiple annotation then the documents will simply be shared out by all annotators and the project will be complete once each document has been annotated by one person.
